argument_parser: doxygen gives me a lot of warnings

I get a ton of warnings building our documentation.

Apparently, `short_id` seems to be the issue.

https://github.com/seqan/seqan3/blob/7fbbefe94b542499ad58bd9293092fe9b84b2a9e/include/seqan3/argument_parser/argument_parser.hpp#L270-L279

I think this is caused by this:

> Note that unlike standard Markdown, doxygen leaves the following untouched.
> 
> >  A `cool' word in a `nice' sentence.
> 
> In other words; a single quote cancels the special treatment of a code span wrapped in a pair of backtick characters. This extra restriction was added for backward compatibility reasons.

https://www.stack.nl/~dimitri/doxygen/manual/markdown.html#mddox_code_spans

complete warning log:
```
warning: Tag `CLANG_ASSISTED_PARSING' at line 1020 of file `Doxyfile' belongs to an option that was not enabled at compile time.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u", or recompile doxygen with this feature enabled.
warning: Tag `CLANG_OPTIONS' at line 1028 of file `Doxyfile' belongs to an option that was not enabled at compile time.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u", or recompile doxygen with this feature enabled.
seqan3/include/seqan3/argument_parser/argument_parser.hpp:280: warning: The following parameters of seqan3::argument_parser::add_flag(bool &value, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec=option_spec::DEFAULT) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/argument_parser.hpp:251: warning: The following parameters of seqan3::argument_parser::add_option(option_type &value, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec=option_spec::DEFAULT, validator_type validator=validator_type{}) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_help.hpp:123: warning: The following parameters of seqan3::detail::format_help::add_flag(bool &, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_help.hpp:101: warning: The following parameters of seqan3::detail::format_help::add_option(option_type &value, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec, validator_type &&validator) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_html.hpp:99: warning: The following parameters of seqan3::detail::format_html::add_flag(bool &, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_html.hpp:77: warning: The following parameters of seqan3::detail::format_html::add_option(option_type &value, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec, validator_type &&validator) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_man.hpp:97: warning: The following parameters of seqan3::detail::format_man::add_flag(bool &, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_man.hpp:75: warning: The following parameters of seqan3::detail::format_man::add_option(option_t &value, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec, validator_type &&validator) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_parse.hpp:138: warning: The following parameters of seqan3::detail::format_parse::add_flag(bool &value, char const short_id, std::string const &long_id, std::string const &, option_spec const &) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_parse.hpp:117: warning: The following parameters of seqan3::detail::format_parse::add_option(option_type &value, char const short_id, std::string const &long_id, std::string const &, option_spec const &spec, validator_type &&validator) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_parse.hpp:649: warning: The following parameters of seqan3::detail::format_parse::get_flag(bool &value, char const short_id, std::string const &long_id) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_parse.hpp:606: warning: The following parameters of seqan3::detail::format_parse::get_option(option_type &value, char const short_id, std::string const &long_id, option_spec const &spec, validator_type &&validator) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_help.hpp:123: warning: The following parameters of seqan3::detail::format_help::add_flag(bool &, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_help.hpp:101: warning: The following parameters of seqan3::detail::format_help::add_option(option_type &value, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec, validator_type &&validator) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_help.hpp:743: warning: The following parameters of seqan3::detail::format_version::add_flag(bool &value, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec) are not documented:
  parameter 'long_id'
seqan3/include/seqan3/argument_parser/detail/format_help.hpp:719: warning: The following parameters of seqan3::detail::format_version::add_option(option_type &value, char const short_id, std::string const &long_id, std::string const &desc, option_spec const &spec, validator_type &&validator) are not documented:
  parameter 'long_id'
```

cc: @smehringer

--- a/include/seqan3/argument_parser/argument_parser.hpp
+++ b/include/seqan3/argument_parser/argument_parser.hpp
@@ -234,8 +234,8 @@ public:
      *                        value. Must satisfy seqan3::validator_concept.
      *
      * \param[out] value     The variable in which to store the given command line argument.
-     * \param[in]  short_id  The short identifier for the option (e.g. `'a'`).
-     * \param[in]  long_id   The long identifier for the option (e.g. `"age"`).
+     * \param[in]  short_id  The short identifier for the option (e.g. 'a').
+     * \param[in]  long_id   The long identifier for the option (e.g. "age").
      * \param[in]  desc      The description of the option to be shown in the help page.
      * \param[in]  spec      Advanced option specification. see seqan3::option_spec.
      * \param[in]  validator The validator applied to the value after parsing (callable).
@@ -270,8 +270,8 @@ public:
     /*!\brief Adds a flag to the seqan3::argument_parser.
      *
      * \param[out] value    The variable in which to store the given command line argument.
-     * \param[in]  short_id The short identifier for the flag (e.g. `'i'`).
-     * \param[in]  long_id  The long identifier for the flag (e.g. `"integer"`).
+     * \param[in]  short_id The short identifier for the flag (e.g. 'i').
+     * \param[in]  long_id  The long identifier for the flag (e.g. "integer").
      * \param[in]  desc     The description of the flag to be shown in the help page.
      * \param[in]  spec     Advanced flag specification. see seqan3::option_spec.
      *

--- a/include/seqan3/argument_parser/detail/format_help.hpp
+++ b/include/seqan3/argument_parser/detail/format_help.hpp
@@ -91,8 +91,8 @@ public:
      * \tparam validator_type The type of validator applied to the value after parsing.
      *
      * \param[out] value     The variable in which to store the given command line argument.
-     * \param[in]  short_id  The short identifier for the option (e.g. `'i'`).
-     * \param[in]  long_id   The long identifier for the option (e.g. `"integer"`).
+     * \param[in]  short_id  The short identifier for the option (e.g. 'i').
+     * \param[in]  long_id   The long identifier for the option (e.g. "integer").
      * \param[in]  desc      The description of the option.
      * \param[in]  spec      Advanced option specification. see seqan3::option_spec.
      * \param[in]  validator The validator applied to the value after parsing (callable).
@@ -115,8 +115,8 @@ public:
 
     /*!\brief Adds a seqan3::print_list_item call to be evaluated later on.
      *
-     * \param[in]  short_id The short identifier for the flag (e.g. `'i'`).
-     * \param[in]  long_id  The long identifier for the flag (e.g. `"integer"`).
+     * \param[in]  short_id The short identifier for the flag (e.g. 'i').
+     * \param[in]  long_id  The long identifier for the flag (e.g. "integer").
      * \param[in]  desc     The description of the flag.
      * \param[in]  spec     Advanced flag specification. see seqan3::option_spec.
      */
@@ -709,8 +709,8 @@ public:
      * \tparam validator_type The type of validator applied to the value after parsing.
      *
      * \param[out] value     The variable in which to store the given command line argument.
-     * \param[in]  short_id  The short identifier for the option (e.g. `'i'`).
-     * \param[in]  long_id   The long identifier for the option (e.g. `"integer"`).
+     * \param[in]  short_id  The short identifier for the option (e.g. 'i').
+     * \param[in]  long_id   The long identifier for the option (e.g. "integer").
      * \param[in]  desc      The description of the option.
      * \param[in]  spec      Advanced option specification. see seqan3::option_spec.
      * \param[in]  validator The validator applied to the value after parsing (callable).
@@ -735,8 +735,8 @@ public:
      *
      * \param[out] value    The variable in which to store the given command line argument
      *                      (which is only needed for the interface here).
-     * \param[in]  short_id The short identifier for the flag (e.g. `'i'`).
-     * \param[in]  long_id  The long identifier for the flag (e.g. `"integer"`).
+     * \param[in]  short_id The short identifier for the flag (e.g. 'i').
+     * \param[in]  long_id  The long identifier for the flag (e.g. "integer").
      * \param[in]  desc     The description of the flag.
      * \param[in]  spec     Advanced flag specification. see seqan3::option_spec.
      */

--- a/include/seqan3/argument_parser/detail/format_html.hpp
+++ b/include/seqan3/argument_parser/detail/format_html.hpp
@@ -67,8 +67,8 @@ public:
      * \tparam validator_type The type of validator applied to the value after parsing.
      *
      * \param[out] value     The variable in which to store the given command line argument.
-     * \param[in]  short_id  The short identifier for the option (e.g. `'i'`).
-     * \param[in]  long_id   The long identifier for the option (e.g. `"integer"`).
+     * \param[in]  short_id  The short identifier for the option (e.g. 'i').
+     * \param[in]  long_id   The long identifier for the option (e.g. "integer").
      * \param[in]  desc      The description of the option.
      * \param[in]  spec      Advanced option specification. see seqan3::option_spec.
      * \param[in]  validator The validator applied to the value after parsing (callable).
@@ -91,8 +91,8 @@ public:
 
     /*!\brief Adds a print_list_item call to be evaluated later on.
      *
-     * \param[in]  short_id The short identifier for the flag (e.g. `'i'`).
-     * \param[in]  long_id  The long identifier for the flag (e.g. `"integer"`).
+     * \param[in]  short_id The short identifier for the flag (e.g. 'i').
+     * \param[in]  long_id  The long identifier for the flag (e.g. "integer").
      * \param[in]  desc     The description of the flag.
      * \param[in]  spec     Advanced flag specification. see seqan3::option_spec.
      */

--- a/include/seqan3/argument_parser/detail/format_man.hpp
+++ b/include/seqan3/argument_parser/detail/format_man.hpp
@@ -65,8 +65,8 @@ public:
     /*!\brief Adds a seqan3::detail::print_list_item call to be evaluated later on.
      *
      * \param[out] value     The variable in which to store the given command line argument.
-     * \param[in]  short_id  The short identifier for the option (e.g. `'i'`).
-     * \param[in]  long_id   The long identifier for the option (e.g. `"integer"`).
+     * \param[in]  short_id  The short identifier for the option (e.g. 'i').
+     * \param[in]  long_id   The long identifier for the option (e.g. "integer").
      * \param[in]  desc      The description of the option.
      * \param[in]  spec      Advanced option specification. see seqan3::option_spec.
      * \param[in]  validator The validator applied to the value after parsing (callable).
@@ -89,8 +89,8 @@ public:
 
     /*!\brief Adds a seqan3::detail::print_list_item call to be evaluated later on.
      *
-     * \param[in]  short_id The short identifier for the flag (e.g. `'i'`).
-     * \param[in]  long_id  The long identifier for the flag (e.g. `"integer"`).
+     * \param[in]  short_id The short identifier for the flag (e.g. 'i').
+     * \param[in]  long_id  The long identifier for the flag (e.g. "integer").
      * \param[in]  desc     The description of the flag (which is only needed for the interface here).
      * \param[in]  spec     Advanced flag specification. see seqan3::option_spec.
      */

--- a/include/seqan3/argument_parser/detail/format_parse.hpp
+++ b/include/seqan3/argument_parser/detail/format_parse.hpp
@@ -105,8 +105,8 @@ public:
      * \tparam validator_type The type of validator applied to the value after parsing.
      *
      * \param[out] value     The variable in which to store the given command line argument.
-     * \param[in]  short_id  The short identifier for the option (e.g. `'i'`).
-     * \param[in]  long_id   The long identifier for the option (e.g. `"integer"`).
+     * \param[in]  short_id  The short identifier for the option (e.g. 'i').
+     * \param[in]  long_id   The long identifier for the option (e.g. "integer").
      * \param[in]  spec      Advanced option specification. see seqan3::option_spec.
      * \param[in]  validator The validator applied to the value after parsing (callable).
      *
@@ -129,8 +129,8 @@ public:
     /*!\brief Adds a get_flag call to be evaluated later on.
      *
      * \param[out] value    The variable in which to store the given command line argument.
-     * \param[in]  short_id The short identifier for the flag (e.g. `'i'`).
-     * \param[in]  long_id  The long identifier for the flag (e.g. `"integer"`).
+     * \param[in]  short_id The short identifier for the flag (e.g. 'i').
+     * \param[in]  long_id  The long identifier for the flag (e.g. "integer").
      *
      * \throws seqan3::parser_design_error
      */
@@ -584,8 +584,8 @@ private:
     /*!\brief Handles command line option retrieval.
      *
      * \param[out] value     The variable in which to store the given command line argument.
-     * \param[in]  short_id  The short identifier for the option (e.g. `'i'`).
-     * \param[in]  long_id   The long identifier for the option (e.g. `"integer"`).
+     * \param[in]  short_id  The short identifier for the option (e.g. 'i').
+     * \param[in]  long_id   The long identifier for the option (e.g. "integer").
      * \param[in]  spec      Advanced option specification. see seqan3::option_spec.
      * \param[in]  validator The validator applied to the value after parsing (callable).
      *
@@ -641,8 +641,8 @@ private:
     /*!\brief Handles command line flags, whether they are set or not.
      *
      * \param[out] value    The variable in which to store the given command line argument.
-     * \param[in]  short_id The short identifier for the flag (e.g. `'i'`).
-     * \param[in]  long_id  The long identifier for the flag (e.g. `"integer"`).
+     * \param[in]  short_id The short identifier for the flag (e.g. 'i').
+     * \param[in]  long_id  The long identifier for the flag (e.g. "integer").
      *
      */
     void get_flag(bool & value,

